jwst/resample/tests/test_resample_step.py

```diff
@@ -493,3 +493,39 @@ def test_custom_wcs_pscale_resample_imaging(nircam_rate, ratio):
 
     # test scales are close
     assert np.allclose(output_scale, input_scale * 0.75)
+
+def test_pixscale(nircam_rate):
+
+    # check that if both 'pixel_scale_ratio' and 'pixel_scale' are passed in,
+    # that 'pixel_scale' overrides correctly 
+    im = AssignWcsStep.call(nircam_rate, sip_approx=False)
+    pixarea = im.meta.photometry.pixelarea_arcsecsq
+
+    # check when both pixel_scale and pixel_scale_ratio are passed in
+    res = ResampleStep.call(im, pixel_scale=0.04, pixel_scale_ratio=0.7)
+    assert np.allclose(res.meta.resample.pixel_scale_ratio, 0.04**2 / pixarea)
+
+    # just pixel_scale
+    res = ResampleStep.call(im, pixel_scale=0.04)
+    assert np.allclose(res.meta.resample.pixel_scale_ratio, 0.04**2 / pixarea)
+
+    # just pixel_scale_ratio
+    res = ResampleStep.call(im, pixel_scale_ratio=0.7)
+    assert res.meta.resample.pixel_scale_ratio == 0.7
```

Comment:
I do not think that tests should be checking [only] that correct value is populated/recorded in the `meta`. I think the test should be checking that output image of this step actually was created taking into account these parameters.